cmd/bucket-metadata.go

```diff
@@ -373,3 +376,44 @@ func deleteBucketMetadata(ctx context.Context, obj objectDeleter, bucket string)
 	}
 	return nil
 }
+
+// migrate config for remote targets by encrypting data if currently unencrypted
+func (b *BucketMetadata) migrateTargetConfig(ctx context.Context, objectAPI ObjectLayer) error {
+	var (
+		data []byte
+		err  error
+	)
+	if len(b.BucketTargetsConfigJSON) == 0 {
+		return nil
+	}
+	// Is rotating of creds requested?
+	var reencrypt bool
+	if globalOldCred.IsValid() {
+		data, err = decryptData(b.BucketTargetsConfigJSON, globalOldCred, globalActiveCred)
+		if err != nil {
+			return err
+		}
+		reencrypt = true
+	} else {
+		data = b.BucketTargetsConfigJSON
+	}
+
+	if !utf8.Valid(data) {
+		if _, err = decryptData(data, globalActiveCred); err != nil {
```

Comment:
`data` is not updated, so it is still encrypted.

If `reencrypt = true` you get double encryption?

While `parseBucketTargetConfig` does double decryption surely that cannot be the intent? Then the oldcreds can never be rotated out?

Reply:
marking this comment resolved - switched to encrypting only if kms is configured